oauth2_http/java/com/google/auth/oauth2/ServiceAccountCredentials.java

```diff
@@ -291,6 +299,49 @@ public static ServiceAccountCredentials fromPkcs8(
         null);
   }
 
+  /**
+   * Factory with minimum identifying information and custom transport using PKCS#8 for the private
+   * key.
+   *
+   * @param clientId client ID of the service account from the console. May be null.
+   * @param clientEmail client email address of the service account from the console.
+   * @param privateKeyPkcs8 RSA private key object for the service account in PKCS#8 format.
+   * @param privateKeyId private key identifier for the service account. May be null.
+   * @param scopes scope strings for the APIs to be called. May be null or an empty collection,
+   *     which results in a credential that must have createScoped called before use.
+   * @param transportFactory HTTP transport factory, creates the transport used to get access
+   *     tokens.
+   * @param tokenServerUri URI of the end point that provides tokens.
+   * @param serviceAccountUser the email of the user account to impersonate, if delegating
+   *     domain-wide authority to the service account.
+   * @param quotaProjectId the project used for quota and billing purposes. May be null.
+   * @return New ServiceAccountCredentials created from a private key.
+   * @throws IOException if the credential cannot be created from the private key.
+   */
+  public static ServiceAccountCredentials fromPkcs8(
```

Comment:
I'm not a huge fan of adding more and more of these helpers for each combination of options. We should consider our options around this design for 1.0.

Reply:
agreed. Maybe use a builder eventually.

Reply:
IMO, if this is not necessary for our use case, let's not add it if we're considering deprecating it soon.

Reply:
fair, will revert these changes.